torch/csrc/jit/codegen/cuda/transform_iter.cpp

```diff
@@ -632,198 +632,112 @@ namespace {
+      active_forwarding_map = &producer_forwarding_map;
+      active_compliment_map = &producer_compliment_map;
+      active_dim_flags = &sop->getSqueezeDimFlags();
+      active_root_dom =
+          TensorDomain::noReductions(producer->getMaybeRFactorDomain());
+      active_tv = producer;
+    } else {
+      return;
     }
 
-    // We have root axes in consumer that don't exist in producer, now forward
-    // those to include all id's in consumer comprised of only axes not in
-    // producer.
-    auto consumer_bcast_ids_not_in_producer =
-        consumer_bcast_roots_not_in_producer;
-
-    std::vector<Expr*> consumer_history = StmtSort::getExprs(
-        FusionGuard::getCurFusion(),
-        std::vector<Val*>(
-            consumer->domain()->domain().begin(),
-            consumer->domain()->domain().end()));
-
-    auto isIdOnlyInConsumer =
-        [&consumer_bcast_ids_not_in_producer](IterDomain* input_id) {
-          return consumer_bcast_ids_not_in_producer.find(input_id) !=
-              consumer_bcast_ids_not_in_producer.end();
-        };
-
-    for (auto expr : consumer_history) {
-      auto input_ids = ir_utils::filterByType<IterDomain>(expr->inputs());
-      // If expr inputs are all in consumer_bcast_ids_not_in_producer, than so
-      // are all outputs
-      if (std::all_of(input_ids.begin(), input_ids.end(), isIdOnlyInConsumer)) {
-        // add all outputs to not being in producer
-        for (auto output_ids :
-             ir_utils::filterByType<IterDomain>(expr->outputs())) {
-          consumer_bcast_ids_not_in_producer.emplace(output_ids);
-        }
-      } else if (
-          expr->isA<Merge>() &&
-          std::any_of(input_ids.begin(), input_ids.end(), isIdOnlyInConsumer)) {
-        auto merge_expr = expr->as<Merge>();
-        // If
-        // - one of the inputs is made of id's in consumer that don't map to
-        // producer (bcast axes),
-        // - && the other input maps to an id in both consumer and producer
-        // - && this is a merge
-        //   for the sake of BestEffortReplay we can forward the input mapping
-        //   to both consumer and producer to the output of the expression
-        std::vector<IterDomain*> forwarded_ids;
-        std::vector<IterDomain*> compliment_ids;
-
-        for (auto input_id : input_ids) {
-          if (!isIdOnlyInConsumer(input_id)) {
-            forwarded_ids.emplace_back(input_id);
-            forwarding_map.emplace(std::make_pair(input_id, merge_expr->out()));
-          } else {
-            compliment_ids.push_back(input_id);
-          }
-        }
+    TORCH_INTERNAL_ASSERT(active_root_dom.size() == active_dim_flags->size());
 
-        // Set up compliment map
-        for (auto forwarded_id : forwarded_ids) {
-          compliment_map.emplace(std::make_pair(forwarded_id, compliment_ids));
-        }
+    // Collect which unique broadcast root axes are in only in active_tv but not
```

Comment:
These comments and variable names are a little confusing as they may not about broadcast IDs but squeezed IDs. Maybe just `forwarded_ids`?

Reply:
changed